lec09-modeling-r.Rmd

```diff
@@ -1,4 +1,615 @@
+
+We could have a treacherous mountain pass:
+
+![Abra Malaga Pass, Peru](https://d36tnp772eyphs.cloudfront.net/blogs/1/2015/06/4a.Peru-Abra-Malaga2.jpg)
+
+Here's an idealized representation of a mountain pass. 
+
+![Idealized mountain pass, By Paulba Legend (talk), CC BY-SA 3.0](https://en.wikipedia.org/wiki/Mountain_pass#/media/File:Saddleroute3.JPG)
+
+#### Challenge
+
+Is the high point of a mountain pass a **stable** fixed point or an **unstable** fixed point? Why?
+
+```{r, eval=FALSE, echo=FALSE}
+# Challenge solution
+
+# a fixed point like a mountain pass is called a 'saddle point', and it's unstable. 
+# It has one stable direction (from the peak down into the pass) and one unstable direction (down from the pass)
+```
+
+### Calculating the Jacobian matrix
+
+We will now learn how to formally describe the stability of fixed points in two-dimensional systems --- 
+how to quantitativley distinguish all the possible landscape shapes that can surround a fixed point.
+
+Just like in the one-dimensional case, we want to calculate the slope of the two-dimensional surface 
+in the vicinity of the fixed point. To do this, we calculate four partial derivatives and arrange them
+into a matrix called the **Jacobian**.
+
+Let's use our predator-prey model as an example. We let the right-hand side of each equation 
+equal $f(x,y)$ and $g(x,y)$ respectively.
+
+$$\frac{dx}{dt} = ax - bxy = f(x,y)$$
+$$\frac{dy}{dt} = cx - dy = g(x,y)$$
+
+This is the Jacobian matrix. It's often called $J$.
+
+$$J=\begin{bmatrix}
+    \frac{\partial f}{\partial x} & \frac{\partial f}{\partial y} \\
+    \frac{\partial g}{\partial x} & \frac{\partial g}{\partial y}
+\end{bmatrix}$$
+
+*Note*: it doesn't matter which variable is which, or which equation is called $f$ or $g$, as 
+long as the order is consistent --- one function on the top row, the other on the bottom, one 
+variable in the first column, the other in the second.
+
+Let's evaluate each of these partial derivatives.
+
+$$\frac{\partial f}{\partial x} = \frac{\partial }{\partial x} (ax-bxy) = a - by$$
+
+Think of the partial derivative as a slice in a particular direction through a potato: you could
+look at how the potato shape changes along its long direction or its short direction, and by doing 
+a partial derivative you are slicing along only one of the directions. 
+
+$$\frac{\partial f}{\partial y} = \frac{\partial }{\partial y} (ax-bxy) = - bx$$
+$$\frac{\partial g}{\partial x} = \frac{\partial }{\partial x} (cx-dy) = c$$
+
+$$\frac{\partial g}{\partial x} = \frac{\partial }{\partial x} (cx-dy) = -d$$
+
+We can put all these terms into the Jacobian. This matrix is now the Jacobian corresponding
+to our predator-prey model.
+
+$$J=\begin{bmatrix}
+    a - by & -bx \\
+    c & -d
+\end{bmatrix}$$
+
+### Classifying fixed points using eigenvalues of the Jacobian
+
+From here, we need to calculate the **eigenvalues** of the Jacobian matrix. The eigenvalues of a
```

Comment:
Can you add a colloquial/graphical explanation of why the eigenvalues will tell us the slope/stability of the fix points?